### PR TITLE
Prevalence patient search

### PR DESCRIPTION
**Pull request by davidbgomes**
### :pushpin: References

-   **Issue:** Closes#8694afm42 [Patient search in Prevalence module](https://app.clickup.com/t/8694afm42)

### :memo: Implementation

- Add Prevelance patient search logic. Since the trackedEntity endpoint doesn't support the rootJunction=OR, in order to filter by one field or the other, I've created two separate calls for each and joined them.

### :video_camera: Screenshots/Screen capture
image

### CODE REVIEW

**Review by MiquelAdell — APPROVED** [review]

**Review by 9sneha-n — CHANGES_REQUESTED** [review]
@davidbgomes If i type "aa" in the search and then delete one char "a" then there are duplicate surveys displayed as seen in screenshot. There should be only 2 surveys, but they are displayed twice 
Screenshot 2024-05-07 at 11 47 49 AM

**Comment by 9sneha-n** [comment]
@davidbgomes based on slack discussion with miquel on amr-surveys channel, we can use api filter correct?
`/api/tracker/trackedEntities?program=i0msBbbQxYC&orgUnit=di7FkWFGyuS&fields=trackedEntity,attributes&filter=tlRPoWumrSa%3Aeq%3AdAGk9nbjMeI&filter=yEkJlUFeJdP%3Alike%3Aaa&rootJunction=AND`
Let me know if you see any issue with this approach.

File: src/webapp/components/survey-list/SurveyList.tsx [hunk 1]
```diff
@@ -82,10 +82,18 @@ export const SurveyList: React.FC<SurveyListProps> = ({ surveyFormType }) => {
                             surveys
                         ) && (
                             <ButtonWrapper>
-                                {surveyFormType === "PPSPatientRegister" && (
+                                {["PPSPatientRegister", "PrevalenceCaseReportForm"].includes(
```
Comment (9sneha-n) [inline src/webapp/components/survey-list/SurveyList.tsx:85]: could you please create an array  like SURVEYS_WITH_FILTER_ENABLED or something similar and use it everywhere.

**Comment by davidbgomes** [comment]
Hi @9sneha-n , addressed the comment about using an array for survey types. As we talked on slack, the endpoints /events and /trackedEntities endpoints no longer work with `rootJunction=OR`, it acts like the default AND. I'll keep it as it for now and we'll see what can be done in that regard.

**Comment by 9sneha-n** [comment]
Fixed in https://github.com/EyeSeeTea/amr-surveys/pull/59

**Pull request closed without merging**